docs/development/setup-advanced.md

```diff
@@ -136,6 +137,116 @@ Store.
+   <other stuff>...
+   ==> default: Waiting for the machine to report its IP address...
+       default: Timeout: 120 seconds
+       default: IP: 172.28.119.70
+   ==> default: Waiting for machine to boot. This may take a few minutes...
+       default: SSH address: 172.28.122.156
+   ==> default: Machine booted and ready!
+   ==> default: Preparing SMB shared folders...
+   Vagrant requires administrator access for pruning SMB shares and
+   may request access to complete removal of stale shares.
+   ==> default: Starting the machine...
+   <other stuff>...
+    default: Username (user[@domain]): <your-machine-username>
+    default: Password (will be hidden):
+   ```
+
+   At this point, you will be prompted for your administrator username and password.
+   This should be your administrator account and password credentials (and not your
+   Microsoft account credentials).
+
+3. Set the `EXTERNAL_HOST` environment variable.
+
+   ```bash
+   (zulip-py3-venv) vagrant@ubuntu-18:/srv/zulip$ export EXTERNAL_HOST="$(hostname -I | xargs):9991"
+   (zulip-py3-venv) vagrant@ubuntu-18:/srv/zulip$ echo $EXTERNAL_HOST
+   ```
+
+   The output will be like:
+
+   ```text
+   172.28.122.156:9991
+   ```
+
+   Make sure you note down this down. This is where your zulip development web
+   server can be accessed.
+
+   ```eval_rst
+   .. important::
+      The output of the above command changes every time you restart the Vagrant
+      development machine. Thus, it will have to be run every time you bring one up.
+      The Zulip developers are currently in the process of deciding the best method
+      to automate this process.
+   ```
+
+4. If all went well, you should now be able to bring up the Zulip development web server.
+
+   ```bash
+   (zulip-py3-venv) vagrant@ubuntu-18:/srv/zulip$ ./tools/run-dev.py
+   ```
+
+   The output will look like:
+
+   ```text
+   Starting Zulip on http://172.28.122.156:9991:9991/.  Internal ports:
+   9991: Development server proxy (connect here)
+   9992: Django
+   9993: Tornado
+   9994: webpack
+   9995: Thumbor
+   ```
+
+   Visit http://172.28.122.156:9991/ on your web browser.
+
+5. You can stop the development environment using `vagrant halt`, and restart it
+   using `vagrant up` and then going through steps **3** and **4** again.
+
+### Problems you may encounter
+
+1. If you get an error saying `Hyper-V could not initialize memory`, you can
+   generally fix it by closing all other running programs and running
+   `vagrant up --provider=hyperv` again. You can reopen the other programs after
+   the provisioning is completed. If it still isn't enough, try restarting your
+   system and running the command again.
+
+2. If you get an error on your browser saying `can't connect to http://172.28.122.156:9991/`,
+   you might have to wait for a couple of minutes before trying again. This problem
+   generally disappears after the first time when you run `./tools/run-dev.py`.
```

Comment:
Hmm -- this feels like not something that I'd expect to just resolve itself by waiting. Or maybe by waiting a few seconds, if `tools/run-dev.py` is just starting up, but once it's had a few seconds to start and is saying it's ready, I wouldn't expect waiting another few minutes to help.

I'm going to guess that when you saw this, and then it started working a bit later, you were trying other things in between and one of those fixed it. :slightly_smiling_face:

One thing that would certainly cause this is if `tools/run-dev.py` simply wasn't running and needed to be started. Another is if you were using the wrong IP address -- like if you had the browser pointed at an address where the dev server had been before, but you'd halted and started the VM since then, and it had some other new IP address now.

I think the advice I'd want to give here is: check those two things; then if it's still not working, come ask in chat and we'll debug.

(Even if there really is some failure mode where waiting a couple of minutes can resolve it, that's something it'd be good to debug so we can fix it!)

Reply:
I provisioned a fresh machine on my system and looks like the first run of `./tools/run-dev.py` takes about a minute or so. This disappears after the next reload of the VM. I changed the instructions accordingly.